js/ui/text_box/text_box.js

```diff
@@ -94,6 +94,20 @@ const TextBox = TextEditor.inherit({
         this._$searchIcon = $searchIcon;
     },
 
+
+    _renderLabel: function() {
+        this.callBase();
+
+        if(this._$searchIcon && this._$label) {
+            const labelBeforeElement = this._$label.find('.dx-label-before');
+            const labelBeforeWidth = labelBeforeElement.width() + this._$searchIcon.outerWidth();
+
+            labelBeforeElement.css('width', labelBeforeWidth);
+
+            this._$label.find('.dx-label').css('max-width', this._input().parent().width() - labelBeforeWidth - this._$label.find('.dx-label-after').width());
```

Comment:
I know now why 'max-len' eslint property exists :). I think a good idea to split this line and add some named variables to know what these calculations do.